src/files-and-uploads/data/slice.js

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable no-param-reassign */
+import { createSlice } from '@reduxjs/toolkit';
+
+import { RequestStatus } from '../../data/constants';
+
+const slice = createSlice({
+  name: 'assets',
+  initialState: {
+    assetIds: [],
+    loadingStatus: RequestStatus.IN_PROGRESS,
+    savingStatus: '',
+    addingStatus: '',
+    deletingStatus: '',
+    errors: {
+      upload: [],
+      delete: [],
+      lock: [],
+    },
+    totalCount: 0,
+  },
+  reducers: {
+    setAssetIds: (state, { payload }) => {
+      state.assetIds = payload.assetIds;
+    },
+    setTotalCount: (state, { payload }) => {
+      state.totalCount = payload.totalCount;
+    },
+    updateLoadingStatus: (state, { payload }) => {
```

Comment:
Did you think at all about bringing the request states framework from FLCC over to course authoring for this kind of stuff? If yes, why did you go with this approach?

Reply:
I am using the patterns that already existed in this repo.